session_srtcp.go

```diff
@@ -112,25 +112,47 @@ func (s *SessionSRTCP) decrypt(buf []byte) error {
 		return err
 	}
 
-	pkt, err := rtcp.Unmarshal(decrypted)
-	if err != nil {
-		return err
-	}
-
-	for _, ssrc := range pkt.DestinationSSRC() {
+	forward := func(ssrc uint32) (closed bool, err error) {
 		r, isNew := s.session.getOrCreateReadStream(ssrc, s, newReadStreamSRTCP)
 		if r == nil {
-			return nil // Session has been closed
+			return true, nil // Session has been closed
 		} else if isNew {
 			s.session.newStream <- r // Notify AcceptStream
 		}
 
 		readStream, ok := r.(*ReadStreamSRTCP)
 		if !ok {
-			return fmt.Errorf("failed to get/create ReadStreamSRTP")
+			return false, fmt.Errorf("failed to get/create ReadStreamSRTP")
 		}
 
 		_, err = readStream.write(decrypted)
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	}
+
+	pkt, err := rtcp.Unmarshal(decrypted)
+	if err != nil {
+		return err
+	}
+
+	cpkt := rtcp.CompoundPacket(pkt)
+	ssrcSet := make(map[uint32]bool)
+	if cpkt.Validate() == nil {
```

Comment:
Why validate here at all? Both ends of this if statement will have the same effect.

Let's let applications do compound packet validation instead.

Reply:
The first branch assumes that the DestinationSSRC for a whole CompoundPacket set applies to the whole set, the second branch creates a superset of all of the DestinationSSRC for each packet in the set

the behavior is slightly different (not sure if that's desireable)

Reply:
I think the second branch is the proper behavior.

Say you receive this compound packet indicating that streams with ssrc1, ssrc2, and ssrc3 are all ending:
&CompoundPacket{
  SR(ssrc1),
  SDES(ssrc1 = 'cname1'),
  BYE(ssrc1, ssrc2, ssrc3)
}

You would want to send that to all three srtcp streams.

Reply:
oh! ok, i'll remove the branch.